icu4c/source/i18n/usearch.cpp

```diff
@@ -562,6 +562,26 @@ inline void initialize(UStringSearch *strsrch, UErrorCode *status)
     strsrch->pattern.defaultShiftSize = 0;
 }
 
+// If the caller provided a character breakiterator we'll return that,
+// otherwise we lazily create the internal break iterator. 
+static UBreakIterator* getBreakIterator(UStringSearch *strsrch, UErrorCode *status)
+{
```

Comment:
Before you do anything else, if U_FAILURE then exit with nullptr.

Reply:
Good point, thanks!